### ISSUE TITLE
corrupted coherence maps after `prep_aria`

### ISSUE THREAD

**Issue by rzinke** [insarlab/MintPy#972]
**Description of the problem**

MintPy appears to shift or corrupt the coherence data for an ARIA data set, likely during `prep_aria.py` processing. I have observed this issue for at least two Sentinel-1 orbital tracks.

I pre-process ARIA products using ARIA-tools (commit 04f965fb53fb6edcf6b3ddb4e9b229f8fab1c788) `ariaExtract.py`. I then run `prep_aria.py` with a template file such as the one attached.

This produced the "inputs" folder with "ifgramStack.h5", as expected. However, the coherence maps in the ifgramStack.h5 file are not correct.

Specifically, the coherence maps in the ARIA "stack/cohStack.vrt" file appear geographically registered, with no NaN values except the frame border. This is shown in the attached file, ARIA_output:

![ARIA_output](https://user-images.githubusercontent.com/48299159/223602249-d60851df-938a-499b-b6a1-339afbc1b775.png)

In contrast, the coherence map extracted for the same date pair appears as shown in the image, MintPy_output:
![MintPy_output](https://user-images.githubusercontent.com/48299159/223602548-9b8df715-229d-47be-944f-301ff2bab729.png)

This is the case for all date pairs, as far as I can tell (I examined several).

**System information**

* Operating system: Linux-Ubuntu
* Python environment: manual
* MintPy version: MintPy version 1.5.1.post16, date 2023-03-07
* Your custom / default template file (if the bug is related to a specific dataset): [Parameters_t121B.txt](https://github.com/insarlab/MintPy/files/10915701/Parameters_t121B.txt)

**Comment by yunjunz** [comment]
hmm... does this data corruption happens to the `unwrapPhase` and `connectComponent` as well? They are sharing almost identical code with `coherence` in `prep_aria.py`.

Based on the two screenshots, I could see the corruption, but could not see the potential shift clearly.

One thing I could think of could be somehow mishandling of float64 to float32 [here](https://github.com/insarlab/MintPy/blob/91b5151e730eeaef27ce6687f4c69bb2589ff8c3/src/mintpy/prep_aria.py#L399), if there is. Could you check the data type of `data`, and try to use the following instead?

```python
f["coherence"][ii,:,:] = np.array(data, dtype=np.float32)
```

If it is still a issue, I would recommend a) debug the `prep_aria.py` yourself, or 2) provide commands for a small test site for me to re-produce your error.

**Comment by rzinke** [comment]
Thanks for the thoughts.
Strangely, the connected components and unwrapped phase come out fine. When I extract the files using save_gdal.py inputs/ifgramStack.h5 -d connectComponent-<date1>_<date2> they look just as expected. When I do the same for coherence, it looks as above.

I replaced line 399 in prep_aria as you suggested, and it had no effect.

Very strange.

**Comment by rzinke** [comment]
I tried processing a subset of the data (first half of 2018). The coherence maps were corrupted in the same manner as the larger data set.

**Comment by rzinke** [comment]
Okay, I've found the source of the strange behavior:
When running `ariaTSsetup.py` I set the multilooking value to 3 (i.e., 3 x 3 3-arcsec pixels = 9-arcsec pixels).
After running the TS setup, I do `gdalinfo | grep "Pixel Size"` for a connected components file:
```
Pixel Size = (0.007501111605158,-0.007500000000000)
```

When I do the same for a coherence file, I get:
```
Pixel Size = (0.002500000000000,-0.002500000000000)
```
that is, one third the size.

ARIA-tools is therefore not properly multilooking the coherence maps since their latest update. I'll raise the bug with them.
